controller/api/destination/watcher/ip_watcher.go

```diff
@@ -93,6 +95,27 @@ func NewIPWatcher(k8sAPI *k8s.API, endpoints *EndpointsWatcher, log *logging.Ent
 		},
 	})
 
+	k8sAPI.Pod().Informer().AddIndexers(cache.Indexers{hostIPIndex: func(obj interface{}) ([]string, error) {
```

Comment:
One subtle thing to note about this indexer is that it doesn't capture `hostNetwork` pods with non-hostPort ports, even though they listen on the host network.  For example, if you have a hostNetwork pod which has a (regular) port 8888, it will not get picked up by this indexer even though it listens on the host IP at port 8888.

Reply:
I think this is a problem because it still leaves us open to IP conflicts in the podIP indexer.

For example, if you have 2 hostNetwork pods scheduled on the same host, the hostIP index would return 0 elements (because they don't expose a hostPort) but then the podIP index would return both.

I think to address this we need to explicitly decide if we want to handle hostNetwork pods in the hostIP indexer or in the podIP indexer:
* If we want to handle them in the hostIP indexer, then for hostNetwork pods, we should tread ALL of their ports as if they were hostPorts.  We would also want to exclude hostNetwork pods in the podIP indexers.
* On the other hand, if we want to handle hostNetwork pods in the podIP indexer, then we need to be able to handle the indexer returning more than 1 pod.  In that case I think we would need to iterate through the pods which are returned and find the one with the port that matches the request.

Of these two options I think the first one is easier to reason about because it avoids the podIP indexer containing a mix of host network and pod network IPs.